src/sage/topology/moment_angle_complex.py

```diff
@@ -0,0 +1,803 @@
+        vertices `\{1, 2, 3, \ldots, m\} =: [m]`. `\mathcal{K}_J` denotes the
+        full subcomplex of `\mathcal{K}`, generated by a set of vertices `J`.
+        The same formula holds true for cohomology groups as well.
+
+        .. SEEALSO::
+
+            :meth:`.cell_complex.GenericCellComplex.homology`
+
+        EXAMPLES::
+
+            sage: Z = MomentAngleComplex([[0,1,2], [1,2,3], [3,0]]); Z
+            Moment-angle complex of Simplicial complex with vertex set
+            (0, 1, 2, 3) and facets {(0, 3), (0, 1, 2), (1, 2, 3)}
+            sage: Z = MomentAngleComplex([[0,1,2], [1,2,3], [3,0]])
+            sage: Z.homology()
+            {0: 0, 1: 0, 2: 0, 3: 0, 4: 0, 5: Z x Z, 6: Z, 7: 0}
+            sage: Z.homology(base_ring=GF(2))
+            {0: Vector space of dimension 0 over Finite Field of size 2,
+             1: Vector space of dimension 0 over Finite Field of size 2,
+             2: Vector space of dimension 0 over Finite Field of size 2,
+             3: Vector space of dimension 0 over Finite Field of size 2,
+             4: Vector space of dimension 0 over Finite Field of size 2,
+             5: Vector space of dimension 2 over Finite Field of size 2,
+             6: Vector space of dimension 1 over Finite Field of size 2,
+             7: Vector space of dimension 0 over Finite Field of size 2}
+           sage: RP = simplicial_complexes.RealProjectivePlane()
+           sage: Z = MomentAngleComplex(RP)
+           sage: Z.homology()
+           {0: 0, 1: 0, 2: 0, 3: 0, 4: 0, 5: Z^10, 6: Z^15, 7: Z^6, 8: C2, 9: 0}
+
+        This yields the same result as creating a cubical complex
+        from this moment-angle complex, and then computing its (co)homology,
+        but that is incomparably slower and is really only possible when
+        the associated simplicial complex is very small::
+
+            sage: Z = MomentAngleComplex([[0,1], [1,2], [2,0]]); Z
+            Moment-angle complex of Simplicial complex with vertex set
+            (0, 1, 2) and facets {(0, 1), (0, 2), (1, 2)}
+            sage: Z.cubical_complex()
+            Cubical complex with 64 vertices and 729 cubes
+            sage: Z.cubical_complex().homology() == Z.homology()
+            True
+
+        Meanwhile, the homology computation used here is quite efficient
+        and works well even with significantly larger underlying simplicial
+        complexes::
+
+            sage: Z = MomentAngleComplex([[0,1,2,3,4,5], [0,1,2,3,4,6],
+            ....:                         [0,1,2,3,5,7], [0,1,2,3,6,8,9]])
+            sage: Z.homology()
+            {0: 0,
+             1: 0,
+             2: 0,
+             3: Z^9,
+             4: Z^17,
+             5: Z^12,
+             6: Z x Z x Z,
+             7: 0,
+             8: 0,
+             9: 0,
+             10: 0,
+             11: 0,
+             12: 0,
+             13: 0,
+             14: 0,
+             15: 0,
+             16: 0,
+             17: 0}
+        """
```

Comment:
Please add a few tests for `dim` being a range and with `reduced=False` showing the difference.